eng/common/TestResources/New-TestResources.ps1

```diff
@@ -108,7 +111,8 @@ trap {
 }
 
 # Enumerate test resources to deploy. Fail if none found.
-$root = [System.IO.Path]::Combine("$PSScriptRoot/../../../sdk", $ServiceDirectory) | Resolve-Path
+$repositoryRoot = "$PSScriptRoot/../../.." | Resolve-Path
+$root = [System.IO.Path]::Combine($repositoryRoot, "sdk", $ServiceDirectory) | Resolve-Path
```

Comment:
Any reason we don't just use Join-Path here instead of Combine?

Reply:
@heaths  ?

Reply:
@danieljurek informed me that we are relying on the corky behavior of Combine when it comes to ignoring paths that are absolute paths (https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netcore-3.1), so If `$ServiceDirectory` is an absolute path it will ignore the earlier parameters and just return that absolute path. I don't particularly care for that behavior because it is so unexpected but we can continue using it here although we should probably add a comment about it.

Reply:
Join-Path doesn't handle canonicalization as well, nor does it always add path separators in various cases. It doesn't use `Path.Combine` under the covers even for the FileSystem provider. I don't recommend using it. It's bitten me quite a few times.